### ISSUE TITLE
Changing password with an username that contains special character does not work

### ISSUE THREAD

**Issue by Arlanet-Emre** [umbraco/Umbraco-CMS#17347]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.5.2

### Bug summary

Trying to change the password of a member with an username that contains a special character like '&' gives the following error:
```
Could not reset user password "Invalid username"
```

### Specifics

![image](https://github.com/user-attachments/assets/47ed80a1-c1c0-46bb-b142-6bbc0b12d6b2)

### Steps to reproduce

1. Navigate to the backoffice and open the `Members` tab.
2. Create a member with an username that contains a special character like `umbraco&test@umbraco.com`.
![image](https://github.com/user-attachments/assets/4de5fab5-e1c3-420b-994f-8153fbfcce80)
3. Press the change password button
4. Enter a valid password 
5. Press `Save` to save member

### Expected result / actual result

The member should be saved, just like when creating the member with an username that contains a special character.

 Instead the save button stays stuck on de loading animation. When navigating to `Log Viewer` a warning shows up, explaining that the member password could not be changed (just like in the first screenshot).

**Comment by bielu** [comment]
@Arlanet-Emre https://github.com/umbraco/Umbraco-CMS/issues/14823 I reported this issue sometime ago and partially fixed it but yeah I didn't touch the of validation to fully fix.

**Comment by Arlanet-Emre** [comment]
Small update thanks to @bielu related issue; the error can be fixed by adding '&' character to the `AllowedUserNameCharacters`. It does not explain the inconsistency between being able to save a username containing a special character, but not resetting the password.

**Comment by bielu** [comment]
@Arlanet-Emre the other ticket mentioned does explain it. It use different validations systems and unless it get unified it will be the case, but it is not that easy to do 🤷

**Comment by Arlanet-Emre** [comment]
@bielu You're right. What I meant was the reason for it being different, but you mentioned it not easy to combine it to one validation system.

**Comment by NguyenThuyLan** [comment]
I was able to reproduce it on v13.5.2, v14.3.1 and v15.1.1
When creating a new member with username = `umbraco&test@umbraco.com`, it returns` Invalid username`, 
When updating a member username from `umbracotest@umbraco.com` to `umbraco&test@umbraco.com`, it can be saved
Then do change the password, and it returns '`PasswordChangeFailed`'
The reason seems to be the validation between Create and Update is not consistent
